openingd/dualopend.c

```diff
@@ -3954,10 +3954,26 @@ static void do_reconnect_dance(struct state *state)
 	       sizeof(last_remote_per_commit_secret));
 
 	/* We always send reconnect/reestablish */
+
+	/* BOLT-e299850cb5ebd8bd9c55763bbc498fcdf94a9567 #2:
+	 *
+	 * - if it has sent `commitment_signed` for an
+	 *   interactive transaction construction but it has
+	 *   not received `tx_signatures`:
+	 *   - MUST set `next_funding_txid` to the txid of that
+	 *     interactive transaction.
+	 *   - otherwise:
+	 *   - MUST NOT set `next_funding_txid`.
+	 */
+	tlvs = tlv_channel_reestablish_tlvs_new(tmpctx);
+	if (!tx_state->remote_funding_sigs_rcvd)
+		tlvs->next_funding = &tx_state->funding.txid;
```

Comment:
Would it make sense to `tal_steal` or `tal` a copy onto `tlvs`? Just so we don't accidentally free the pointee.